### PR TITLE
avoid duplicate boost headers for windows

### PR DESCRIPTION
**Pull request by hkiel**
### Related Issues

#7776 

### Purpose

Avoid spreading boost headers everywhere.

### Approach

For gcc use headers in `msys/mingw64/include/boost`, for msvc use `include/omc/3rdParty/boost/msvc/boost`.
Maybe MSVC can be dropped completely?

### CODE REVIEW

**Comment by hkiel** [comment]
@adrpo Can you produce an installable package with this request and try if omsicpp and cpp still work fine? I do not know how to handle those runtimes.

File: OMCompiler/SimulationRuntime/OMSICpp/runtime/src/Core/Modelica/CMakeLists.txt
Comment (mahge) [inline OMCompiler/SimulationRuntime/OMSICpp/runtime/src/Core/Modelica/CMakeLists.txt:124]: @hkiel you have a typo here. wingw64 should be mingw64.

**Comment by hkiel** [comment]
@mahge Can we dare to merge this PR? Can you check if the new nightly for windows has working cpp/omsicpp runtime?
Maybe this can go into 1.19.1 if it works - it should reduce installation time on windows significantly.

**Comment by mahge** [comment]
I think we can merge it and see if the nightly works fine tomorrow. We can also check on the [Windows testing](https://test.openmodelica.org/jenkins/blue/organizations/jenkins/Windows%2FOM_Win/activity) which will run once this is merged. If tests fail on Windows we can check and fix them. 

We can wait a bit for @adrpo and merge it later today.

**Review by mahge — APPROVED** [review]

**Comment by adrpo** [comment]
> @adrpo Can you produce an installable package with this request and try if omsicpp and cpp still work fine? I do not know how to handle those runtimes.

We should just merge it and test it with the nightly builds.

**Comment by hkiel** [comment]
> > @adrpo Can you produce an installable package with this request and try if omsicpp and cpp still work fine? I do not know how to handle those runtimes.
> 
> We should just merge it and test it with the nightly builds.

@adrpo Installation is now 120s faster (uninstall 80s faster). Can you please check if cpp/omsicpp runtimes still work with the unified includes?

**Pull request merged**
